II - Data processing/slides/pandas-data-processing-CE.tex

```diff
@@ -0,0 +1,1181 @@
+
+	\textbf{Column indexing}
+
+	The difference between the two is that the first method doesn't allow column name references, while the second does
+	\begin{figure}
+		\centering
+		\includegraphics[width=0.47\linewidth]{img/col_name_reference.png}
+	\end{figure}
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Multi-column indexing}
+
+	\begin{itemize}
+		\item We can use a syntax similar to the second method to index more than one column at the same time
+		\item Instead of including a string with one column name inside the brackets, we include a list of strings with the column names to index
+	\end{itemize}
+
+	\hspace{7mm} \texttt{df[['col\_name1', 'col\_name2', 'col\_name3', ...]]}
+
+	\begin{itemize}
+		\item Note that inside the outer brackets we have a list of strings
+	\end{itemize}
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Multi-column indexing}
+
+	\begin{multicols}{2}
+
+		\begin{figure}
+			\centering
+			\includegraphics[width=0.8\linewidth]{img/multicol_subsetting1.png}
+		\end{figure}
+		\begin{figure}
+			\centering
+			\includegraphics[width=0.75\linewidth]{img/multicol_subsetting2.png}
+		\end{figure}
+
+	\end{multicols}
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Multi-column indexing}
+
+	A one-column index operation returns a Pandas series. A multicolumn index returns another dataframe.
+
+	\begin{multicols}{2}
+
+		\begin{figure}
+			\centering
+			\includegraphics[width=0.7\linewidth]{img/price.png}
+		\end{figure}
+		\begin{figure}
+			\centering
+			\includegraphics[width=0.7\linewidth]{img/price_quantity.png}
+		\end{figure}
+
+	\end{multicols}
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Row indexing}
+
+	There are basically two methods to index rows in Pandas. The simplest is \texttt{.iloc[]}, which is used to index the i-th row or rows:
+
+	\begin{itemize}
+		\item Indexing a single row: \texttt{df.iloc[i]}
+		\item Indexing a range of continuous rows from i until (j-1): \texttt{df.iloc[i:j]}
+		\item Indexing the i-th and j-th non-continuous rows: \texttt{df.iloc[[i, j]]}
+	\end{itemize}
+
+	\textbf{Very important:} In Python, every numeric index starts at zero, not at one
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Row indexing}
+
+	\begin{itemize}
+		\item Indexing a single row: \texttt{df.iloc[i]}
+	\end{itemize}
+
+	\begin{figure}
+		\centering
+		\includegraphics[width=0.6\linewidth]{img/single_row.png}
+	\end{figure}
+
+	Note that the outcome of indexing a single row is a Pandas series, not a dataframe
+
+\end{frame}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Slide
+\begin{frame}{Indexing and filtering}
+
+	\textbf{Row indexing}
```

Comment:
On this slide or next can you explain when if ever the index update when you subset a pd df? This is something I have wondered.

Reply:
I'm adding a sentence noting it and saying that we'll explain it later

Reply:
added in e2b3b81484ac420ab2f66626b1a8e0a8c9a0964a